### PR TITLE
Fetch additional package metadata

### PR DESCRIPTION
**Pull request by elfenlaid**
Additional fields are:

- `ownerAvatarUrl` for #975
- `ownerAvatarName` as referenced [in the discussion](https://github.com/SwiftPackageIndex/SwiftPackageIndex-Server/pull/1116#issuecomment-845024446)
- `isInOrganization` for #980

From my understanding of Github's GraphQL API, the `Repository.owner` field is shared by both `User` and `Organization` entities. E.g., if a repository belongs to an organization, `login` and `avatarUrl` should point to the organization's metadata. Otherwise, `login` and `avatarUrl` point to the author's account. ([Explorer configuration for API probing](https://gist.github.com/elfenlaid/61d5a72a20278eca948e6e9ff2a5b62c))

So, it's not clear only from the `owner` field whether the author is an organization or a person. Thus, the additional `Repository.isInOrganization` field helps to distinguish the owner's type.

Probably I've missed something (maybe you have a running revamp plan of Repo's ownership), so looking forward to the feedback :)

### CODE REVIEW

File: Sources/App/Core/Github.swift [hunk 1]
```diff
@@ -259,6 +259,13 @@ extension Github {
                     }
                     owner {
                       login
+                      avatarUrl
+                      ... on User {
+                        name
+                      }
+                      ... on Organization {
+                        name
+                      }
```
Comment (finestructure) [inline Sources/App/Core/Github.swift:268]: Interesting, I had no idea there's an `...` operator!

I'm guessing this is sort of like an enum or union type for `name`. Is there a guarantee in the schema that there'll always be either a `User` or an `Organisation`? Or do we potentially need to make this `String?` in our `Owner` `Decodable`?

The graphql console isn't working for me at the moment so I can't check unfortunately.

File: Sources/App/Core/Github.swift [hunk 2]
```diff
@@ -335,6 +344,8 @@ extension Github {
 
         struct Owner: Decodable, Equatable {
             var login: String
+            var name: String
+            var avatarUrl: String
```
Comment (finestructure) [inline Sources/App/Core/Github.swift:348]: Just a reminder to confirm these two are `String!` in the GraphQL schema.

File: Sources/App/Migrations/027/UpdateRepositoryAddOwnerFields.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import Fluent
+
+
+struct UpdateRepositoryAddOwnerNameAndOwnerAvatarUrlAndIsInOrganizationFlag: Migration {
```
Comment (finestructure) [inline Sources/App/Migrations/027/UpdateRepositoryAddOwnerFields.swift:4]: Nit: I'd keep file and struct name in sync, i.e. just call this `UpdateRepositoryAddOwnerFields` (I don't think it collides with any previous name). In hindsight we should probably have namespaces these with `Migration001` etc.

**Review by finestructure — APPROVED** [review]
Looks great, thanks @elfenlaid !

Just added a couple of comments and questions but otherwise 👍

File: Sources/App/Core/Github.swift (hunk 1 above)
Comment (elfenlaid) [reply Sources/App/Core/Github.swift:268]: TL;DR:

- `avatarUrl` and `login` are safe to leave as `String`
- `name` should be `String?`

Oh, there're two instances of `...` in GraphQL I'm aware of:

- [GraphQL Fragments](https://graphql.org/learn/queries/#fragments) - they preserve query's locality
- [GraphQL Interfaces](https://graphql.org/learn/schema/#interfaces) and [GraphQL Unions](https://graphql.org/learn/schema/#union-types) - `... on Organization` is similar to `if let organization = owner as? Organization` to my understanding

So, long story short, there's `interface RepositoryOwner` and two instances that conform to it: `User` and `Organization`. `Repository` declares `owner: RepositoryOwner!`

- `avatarUrl: URI!` and `login: String!` are the `RepositoryOwner` interface part. Thus guaranteed to be presented
- While  `User` and `Organization` both contain the` name: String` parameter, it's not part of the `RepositoryOwner` interface

The query means "Hey if the owner typecasts to User or Organization types, get its name":

```graphql
... on User {
    name
}
... on Organization {
    name
}
```

File: Sources/App/Migrations/027/UpdateRepositoryAddOwnerFields.swift (hunk 3 above)
Comment (elfenlaid) [reply Sources/App/Migrations/027/UpdateRepositoryAddOwnerFields.swift]: > I'd keep file and struct name in sync 

Oops 😳

> (I don't think it collides with any previous name)

Presumable the unique index on `_fluent_migrations.name` should fail if names collide

> In hindsight we should probably have namespaced these with Migration001 etc.

Yeah, maybe [timestamps](https://github.com/hexpm/hexpm/tree/main/priv/repo/migrations ) instead of the incremental counter should work as well. Counters are more prone to conflicts.

**Comment by finestructure** [comment]
Closing and re-opening to make the tests run

**Comment by finestructure** [comment]
Superseded by #1127
